br/pkg/lightning/backend/external/writer.go

```diff
@@ -427,3 +430,39 @@ func (w *Writer) createStorageWriter(ctx context.Context) (
 	}
 	return dataPath, statPath, dataWriter, statsWriter, nil
 }
+
+// EngineWriter implements backend.EngineWriter interface.
+type EngineWriter struct {
+	w *Writer
+}
+
+// NewEngineWriter creates a new EngineWriter.
+func NewEngineWriter(w *Writer) *EngineWriter {
+	return &EngineWriter{w: w}
+}
+
+// AppendRows implements backend.EngineWriter interface.
+func (e *EngineWriter) AppendRows(ctx context.Context, _ []string, rows encode.Rows) error {
+	kvs := kv.Rows2KvPairs(rows)
+	if len(kvs) == 0 {
+		return nil
+	}
+	for _, item := range kvs {
+		err := e.w.WriteRow(ctx, item.Key, item.Val, nil)
```

Comment:
```suggestion
		err := e.w.WriteRow(ctx, item.Key, item.Val, item.RowID)
```

Reply:
`e.w` requires a `handle`, import into don't have conflict detect now, we can do it later.